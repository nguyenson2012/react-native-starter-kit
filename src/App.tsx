/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/emin93/react-native-template-typescript
 *
 * @format
 */

import React, { Fragment, useEffect, useState } from 'react';
import { SafeAreaView, StyleSheet, ScrollView, View, Text, StatusBar, Platform } from 'react-native';
import { Header, Colors } from 'react-native/Libraries/NewAppScreen';
import SplashScreen from 'react-native-splash-screen';
import { config } from '@app/config';
import { Icon, Button } from '@app/components';

const styles = StyleSheet.create({
  scrollView: { backgroundColor: Colors.lighter },
  engine: {
    position: 'absolute',
    right: 0,
  },
  body: { backgroundColor: Colors.white },
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
    color: Colors.black,
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
    color: Colors.dark,
  },
  highlight: { fontWeight: '700' },
  footer: {
    color: Colors.dark,
    fontSize: 12,
    fontWeight: '600',
    padding: 4,
    paddingRight: 12,
    textAlign: 'right',
  },
});

const App = (): JSX.Element => {
  const [effectLoaded, setEffectLoaded] = useState(false);
  const [showHello, setShowHello] = useState(false);

  useEffect((): void => {
    setEffectLoaded(true);
    SplashScreen.hide();
  }, []);

  const onPressHello = (): void => setShowHello(true);

  return (
    <Fragment>
      <StatusBar barStyle='dark-content' />
      <SafeAreaView>
        <ScrollView contentInsetAdjustmentBehavior='automatic' style={styles.scrollView}>
          <Header />
          {(global as any).HermesInternal == null ? null : ( // eslint-disable-line @typescript-eslint/no-explicit-any
            <View style={styles.engine}>
              <Text style={styles.footer}>Engine: Hermes</Text>
            </View>
          )}
          <View style={styles.body}>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>Effect loaded: {effectLoaded ? 'true' : 'false'}</Text>
            </View>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle} testID=''>
                Icon
                <Icon name='home' />
                <Icon ios='ios-menu' android='md-menu' name='home' style={{ fontSize: 20, color: 'red' }} />
                <Icon type='FontAwesome' name='home' />
              </Text>
            </View>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle} testID=''>
                Version: {Platform.OS === 'android' ? config.android.version : config.ios.version}
              </Text>
            </View>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>Platform: {Platform.OS}</Text>
            </View>
            <Button success onPress={onPressHello}>
              <Text> Hello </Text>
            </Button>
            {showHello && (
              <View style={styles.sectionContainer}>
                <Text style={styles.sectionTitle}>Hello world</Text>
              </View>
            )}
          </View>
        </ScrollView>
      </SafeAreaView>
    </Fragment>
  );
};

export default App;
