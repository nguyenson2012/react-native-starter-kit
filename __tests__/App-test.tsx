import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import App from '@app/App';
import { Platform } from 'react-native';

jest.mock('react-native-splash-screen', () => ({
  hide: jest.fn(),
}));

describe('<App/>', () => {
  beforeEach(() => {
    Platform.OS = 'ios';
    (global as any).HermesInternal = undefined;
  });
  it('renders successfully with effect', async () => {
    const { getByText, baseElement } = render(<App />);

    expect(getByText('Effect loaded: true')).toBeDefined();
    // const famousWomanInHistory = 'Ada Lovelace';

    // const input = getByTestId('input');
    // fireEvent.changeText(input, famousWomanInHistory);

    // const button = getByText('Print Username');
    // fireEvent.press(button);

    // await wait(() => expect(queryByTestId('printed-username')).toBeTruthy());

    // expect(getByTestId('printed-username').props.children).toBe(famousWomanInHistory);
    expect(baseElement).toMatchSnapshot();
  });

  it('renders correctly on android platform', async () => {
    Platform.OS = 'android';
    const { getByText, baseElement } = render(<App />);

    expect(getByText('Platform: android')).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });

  it('renders correctly with hermes engine', async () => {
    (global as any).HermesInternal = true;
    const { getByText, baseElement } = render(<App />);

    expect(getByText('Engine: Hermes')).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });
  it('shows "Hellow world" when clicking Hello button', async () => {
    const { getByText } = render(<App />);
    const button = getByText('Hello');
    fireEvent.press(button);

    expect(getByText('Hello world')).toBeDefined();
  });
});
