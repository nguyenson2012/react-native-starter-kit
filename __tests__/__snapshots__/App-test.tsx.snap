// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders correctly on android platform 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView>
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={
          Object {
            "backgroundColor": "#F3F3F3",
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "#F3F3F3",
              "paddingBottom": 40,
              "paddingHorizontal": 32,
              "paddingTop": 96,
            }
          }
        >
          <Image
            accessibilityRole="image"
            source={
              Object {
                "testUri": "../Libraries/NewAppScreen/components/logo.png",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "marginBottom": -192,
                  "marginLeft": -128,
                  "opacity": 0.2,
                  "overflow": "visible",
                  "resizeMode": "cover",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 40,
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
          >
            Welcome to React
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
            }
          }
        >
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              Effect loaded: 
              true
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
              testID=""
            >
              Version: 
              0.2.2
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              Platform: 
              android
            </Text>
          </View>
          <Button
            title="Hello"
          />
        </View>
      </ScrollView>
    </SafeAreaView>
  </View>
</View>
`;

exports[`<App/> renders correctly with hermes engine 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView>
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={
          Object {
            "backgroundColor": "#F3F3F3",
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "#F3F3F3",
              "paddingBottom": 40,
              "paddingHorizontal": 32,
              "paddingTop": 96,
            }
          }
        >
          <Image
            accessibilityRole="image"
            source={
              Object {
                "testUri": "../Libraries/NewAppScreen/components/logo.png",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "marginBottom": -192,
                  "marginLeft": -128,
                  "opacity": 0.2,
                  "overflow": "visible",
                  "resizeMode": "cover",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 40,
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
          >
            Welcome to React
          </Text>
        </View>
        <View
          style={
            Object {
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#444",
                "fontSize": 12,
                "fontWeight": "600",
                "padding": 4,
                "paddingRight": 12,
                "textAlign": "right",
              }
            }
          >
            Engine: Hermes
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
            }
          }
        >
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              Effect loaded: 
              true
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
              testID=""
            >
              Version: 
              0.2.2
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              Platform: 
              ios
            </Text>
          </View>
          <Button
            title="Hello"
          />
        </View>
      </ScrollView>
    </SafeAreaView>
  </View>
</View>
`;

exports[`<App/> renders successfully with effect 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView>
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={
          Object {
            "backgroundColor": "#F3F3F3",
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "#F3F3F3",
              "paddingBottom": 40,
              "paddingHorizontal": 32,
              "paddingTop": 96,
            }
          }
        >
          <Image
            accessibilityRole="image"
            source={
              Object {
                "testUri": "../Libraries/NewAppScreen/components/logo.png",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "marginBottom": -192,
                  "marginLeft": -128,
                  "opacity": 0.2,
                  "overflow": "visible",
                  "resizeMode": "cover",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 40,
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
          >
            Welcome to React
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
            }
          }
        >
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              Effect loaded: 
              true
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
              testID=""
            >
              Version: 
              0.2.2
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              Platform: 
              ios
            </Text>
          </View>
          <Button
            title="Hello"
          />
        </View>
      </ScrollView>
    </SafeAreaView>
  </View>
</View>
`;
